### ISSUE TITLE
File menu takes around 12 seconds to open on laptop with DVD drive

### ISSUE THREAD

**Issue by jhertel** [clsid2/mpc-hc#1205]
I have a laptop with a DVD drive. Most of the times when I try to open the File menu in MPC-HC, it takes around 12 seconds for the menu to appear, during which time MPC-HC is frozen and doesn't respond. Even the entire Windows graphical interface seems partly locked, in that I can't switch to other programs on the task bar, and Alt-Tab does not respond either (nothing happens when I press Alt-Tab). In the waiting time, I can hear sounds from the DVD drive (which has no disc in it). So presumably MPC-HC is trying to access it before displaying the menu – and it does that solely because I want to open the File menu! This is of course intolerable, as the File menu should appear instantaneously, no matter what configuration the computer has or what state the drives are in. MPC-HC shouldn't really access any paths just because the user wants to open a menu, and it definitely shouldn't block almost the entire Windows GUI. Menus should always open instantaneously.

I have not played any DVDs from the drive with MPC-HC, and there is not even an old recent path pointing to it.

Only when I have been opening the File menu very recently (within a minute or a few minutes) does the File menu appear instantaneously.  It doesn't matter whether I close MPC-HC in between these attempts; if it tried to access the drive within a minute or so earlier, and I close MPC, and open it again, the File menu still appears instantaneously. But if I wait a few minutes, even when keeping MPC open, it again takes around 12 seconds for it to open the File menu.

When I try to access the DVD drive in other programs, they _immediately_ tell me there is no disc in the drive. So even if MPC-HC wants to access the DVD drive, it could immediately be told by Windows that there is in fact no drive there; there seems to be no need for MPC to try to read from a non-existent disk.

The DVD drive is drive D: on my laptop, if that matters (I guess it shouldn't).

Operating system: Windows 10 version 21H1 (build 19043.1083).

Version info from MPC-HC:

"MPC-HC (64-bit)
---------------

Build information:
    Version:            1.9.14 (4488e7d5a)
    Compiler:           MSVC v19.26.28806
    Build date:         Jul  2 2021

LAV Filters:
    LAV Splitter:       0.75.1.0
    LAV Video:          0.75.1.0
    LAV Audio:          0.75.1.0
    FFmpeg compiler:    MinGW-w64 GCC 9.3.0

Operating system:
    Name:               Windows NT 10.0 (build 19043)
    Version:            10.0 (64-bit)

Hardware:
    CPU:                Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz
    GPU:                Intel(R) HD Graphics 520 (driver version: 21.20.16.4550)"

**Comment by clsid2** [comment]
Test if this build fixes the delay:
https://www.sendspace.com/file/3cszli

**Comment by jhertel** [comment]
Thanks for the quick reply! 

It took a while to test it, because I was struggling with Windows Defender saying the file contains a "Trojan:Script/Wacatac.B!ml" virus, and therefore blocking the download, and [VirusTotal saying AutoShun saying the same about the zip file](https://www.virustotal.com/gui/url/146a12616eedc9550a5b43d870798f89af6750f6696439cb883886c70d9ab0d9/detection). 

But in the end I managed to download it, unpack the mpc-hc64.exe, [test the mpc-hc64.exe with VirusTotal](https://www.virustotal.com/gui/file/b229025be6ad57cff25b78df32c61175d89f990439e161a1b2772ab13c3d3ef1/detection), which was now all green.

And when I ran this build, I could open the File menu without any problems. It opened instantaneously. Yay! 

For comparison, when I ran the installed MPC-HC right after, the same 12 second freeze happened again in the installed version, as expected.

So that build seems to solve the issue! 🙂  I have never seen such a quick bug fix before! Thank you! 

Can I use that build until the official bug fix comes out?
